superset/views/utils.py

```diff
@@ -207,35 +207,33 @@ def get_time_range_endpoints(
     form_data: Dict[str, Any], slc: Optional[models.Slice]
 ) -> Optional[Tuple[TimeRangeEndpoint, TimeRangeEndpoint]]:
     """
-    Get the slice aware time range endpoints falling back to the SQL database specific
-    definition or default if not defined.
+    Get the slice aware time range endpoints from the form-data falling back to the SQL
+    database specific definition or default if not defined.
 
     For SIP-15 all new slices use the [start, end) interval which is consistent with the
-    Druid REST API.
+    native Druid connector.
 
     :param form_data: The form-data
     :param slc: The chart
     :returns: The time range endpoints tuple
     """
 
-    time_range_endpoints = form_data.get("time_range_endpoints")
+    endpoints = form_data.get("time_range_endpoints")
 
-    if time_range_endpoints:
-        return time_range_endpoints
+    if slc and not endpoints:
```

Comment:
This is really just a refactor of the old logic to ensure that we're not repeating ourselves when defining the interval.